src/providers/packet/mod.rs

```diff
@@ -117,83 +118,103 @@ fn get_dns_servers() -> Result<Vec<IpAddr>> {
 
 fn parse_network(netinfo: &PacketNetworkInfo) -> Result<(Vec<Interface>,Vec<Device>)> {
     let mut interfaces = Vec::new();
+    let mut bonds = Vec::new();
     for i in netinfo.interfaces.clone() {
         let mac = MacAddr::from_str(&i.mac)
             .map_err(|err| Error::from(format!("{:?}", err)))
             .chain_err(|| format!("failed to parse mac address: '{}'", i.mac))?;
         interfaces.push(Interface {
             mac_address: Some(mac),
-            bond: Some("bond0".to_owned()),
+            bond: i.bond.clone(),
             name: None,
             priority: None,
             nameservers: Vec::new(),
             ip_addresses: Vec::new(),
             routes: Vec::new(),
         });
+
+        // if there is a bond key, make sure we have a bond device for it
+        if let Some(ref bond_name) = i.bond {
+            let bond = Interface {
+                name: Some(bond_name.clone()),
+                priority: Some(5),
+                nameservers: get_dns_servers()?,
+                mac_address: Some(mac),
+                bond: None,
+                ip_addresses: Vec::new(),
+                routes: Vec::new(),
+            };
+            if !bonds.contains(&bond) {
+                bonds.push(bond);
+            }
+        }
+    }
+
+    // according to the folks from packet, all the addresses given to us in the
+    // network section should be attached to the first bond we find in the list
+    // of interfaces. there probably won't be no bonds in the list, but if we
+    // don't find any, we just print out a scary warning and don't attach the
+    // addresses to anything.
+    if bonds.is_empty() {
+        warn!("no bond interfaces. addresses are left unassigned.");
+        // the rest of the function operates on bonds, so just return
+        return Ok((interfaces, vec![]));
     }
-    let mut iface = Interface{
-        name: Some("bond0".to_owned()),
-        priority: Some(5),
-        nameservers: get_dns_servers()?,
-        mac_address: None,
-        bond: None,
-        ip_addresses: Vec::new(),
-        routes: Vec::new(),
-    };
+
+    // remove panics if the index is out of bounds, but we know that there is at
+    // least one bond in the vector because we return if it's empty
+    let mut bond0 = bonds.remove(0);
```

Comment:
There's nothing requiring it to actually be named `bond0`, so maybe e.g. `first_bond` would be more descriptive.